new ArgParse code does not build with VS2015

After commit e81036b6559 (#2531), I can no longer build on Windows using VS2015 because I get build errors of the form:
```
oiio\src\oiiotool\oiiotool.cpp(5244): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5244): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5244): note: No constructor could take the source type, or constructor overload resolution was ambiguous
```
 Looking at the code, I can't see how this could be considered ambiguous and this should work, so I suspect a compiler bug. And in fact, if I build with VS2019 I get no error.

I'm hoping very soon we'll be building OIIO with a newer compiler than VS2015, so in my particular case I don't think I need this fixed in OIIO. However, currently the minimum visual studio required, according to the following is 2015:
```
src/include/OpenImageIO/platform.h
# if _MSC_VER < 1900
 # error "This version of OIIO is meant to work only with Visual Studio 2015 or later"
# endif
```
So I think either we'd want to raise the minimum VS (I haven't tested 2017, but I do know 2019 works) or implement a temporary work-around? There's also the possibility I'm messing something up, so of course would be good if someone else could do a sanity check?

Sorry about that, I had some debug code. After reverting that hopefully this will line up better.  Just to be sure, the first error I believe corresponds to:

```
    ap.arg("--info")
      .help("Print resolution and basic info on all inputs, detailed metadata if -v is also used (options: format=xml:verbose=1)")
      .action(set_printinfo);
```

here's the full list of errors for e81036b.:
```
oiio\src\oiiotool\oiiotool.cpp(5652): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5652): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5652): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5655): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5655): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5655): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5664): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5664): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5664): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5669): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5669): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5669): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5672): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5672): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5672): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5681): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5681): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5681): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5700): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5700): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5700): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5703): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(OpenImageIO_v2_2::span<const char *const ,-1>)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5703): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5703): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5724): error C2664: 'OpenImageIO_v2_2::ArgParse::Arg &OpenImageIO_v2_2::ArgParse::Arg::action(int (__cdecl *)(int,const char **))': cannot convert argument 1 from 'void (__cdecl *)(void)' to 'OpenImageIO_v2_2::ArgParse::ArgAction &&'
oiio\src\oiiotool\oiiotool.cpp(5724): note: Reason: cannot convert from 'overloaded-function' to 'OpenImageIO_v2_2::ArgParse::ArgAction'
oiio\src\oiiotool\oiiotool.cpp(5724): note: No constructor could take the source type, or constructor overload resolution was ambiguous
oiio\src\oiiotool\oiiotool.cpp(5725): error C2228: left of '.hidden' must have class/struct/union
```

I'm also stuck at this issue right now with toolset v140
I also have another error in addition to the above ones...
```
oiio\src\include\OpenImageIO/argparse.h(591): error C2075: 'OpenImageIO_v2_2::ArgParse::store_const::<lambda_c281d3b71724d89151a177c99914fd60>::value': array initialization requires a brace-enclosed initializer list
oiio\src\libutil\argparse_test.cpp(181): note: see reference to function template instantiation 'OpenImageIO_v2_2::ArgParse::ArgAction OpenImageIO_v2_2::ArgParse::store_const<char[8]>(const T (&))' being compiled with
          [
              T=char [8]
          ]
```

I also get this warning:
```
oiio\src\libutil\argparse_test.cpp(229): warning C4805: '==': unsafe mix of type 'int' and type 'bool' in operation
```

I found a simple solution to the last error (C2228) that @ThiagoIze reported:
https://github.com/OpenImageIO/oiio/blob/4fc8ce3a29981538fc4fd64b10a8109b2c2cae36/src/oiiotool/oiiotool.cpp#L5309-L5311
The .hidden just needs to be before .action to work, as it's working that way in many other places of the code.
 ```cpp
    ap.arg("--crash")
      .hidden()
      .action(crash_me);
```

The other issues I have no idea how to fix.

--- a/src/oiiotool/oiiotool.cpp
+++ b/src/oiiotool/oiiotool.cpp
@@ -5307,8 +5307,8 @@ getargs(int argc, char* argv[])
     ap.arg("--metamerge", &ot.metamerge)
       .help("Always merge metadata of all inputs into output");
     ap.arg("--crash")
-      .action(crash_me)
-      .hidden();
+      .hidden()
+      .action(crash_me);
     ap.separator("Commands that read images:");
     ap.arg("-i %s:FILENAME")
       .help("Input file (options: now=, printinfo=, autocc=, type=, ch=)")

